src/backend/api/meals.js

```diff
@@ -1,14 +1,79 @@
-const express = require("express");
+const express = require('express');
 const router = express.Router();
-const knex = require("../database");
+const knex = require('../database');
 
-router.get("/", async (request, response) => {
+router.get('/', async (req, res) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const allMeals = await knex.select().table('meal');
+
+    if (allMeals.length !== 0) {
+      res.send(allMeals);
+    } else {
+      res.status(404).json({ error: 'Meals not found' });
+    }
+  } catch (error) {
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+router.post('/', async (req, res) => {
+  try {
+    const newMeal = await knex.insert(req.body).into('meal');
+
+    res.send({ massage: `Meal with ID ${newMeal} created` });
+  } catch (error) {
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+router.get('/:id', async (req, res) => {
+  try {
+    const mealId = req.params.id;
+    const meal = await knex.select().from('meal').where({ id: mealId });
+
+    if (meal.length !== 0) {
+      res.send(meal);
+    } else {
+      res.status(404).json({ error: 'Meal not found' });
+    }
+  } catch (error) {
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+router.put('/:id', async (req, res) => {
+  try {
+    const mealBody = req.body;
+    const mealId = req.params.id;
```

Comment:
Nice one, I get an idea from your code to write dynamic meal id as I hardcode my code 😅